src/spanningtrees/prim.jl

```diff
@@ -1,60 +1,84 @@
 immutable PrimHeapEntry{T<:Real}
+    vertex::Int
     edge::Edge
     dist::T
 end
 
 isless(e1::PrimHeapEntry, e2::PrimHeapEntry) = e1.dist < e2.dist
 
-"""Performs [Prim's algorithm](https://en.wikipedia.org/wiki/Prim%27s_algorithm)
+"""
+Checks whether a given vertex `v` is a member of a Vector{PrimHeapEntry}
+"""
+function contains{T<:Real}(pq::Vector{PrimHeapEntry{T}}, v::Int) :: Bool
+    for item in pq
+        v == item.vertex && return true
+    end
+    return false
+end
+
+"""
+Takes a Vector{PrimHeapEntry} `pq` and a vertex `v` as arguments and returns the
+dist key of the vertex.
+"""
+function get_dist{T<:Real}(pq::Vector{PrimHeapEntry{T}}, v::Int) :: T
+    for item in pq
+        v == item.vertex && return item.dist
```

Comment:
Integers,  dont have a concept of infinity. Complex and Rational do, as do floats. Since you've got T <: Real, you're including integers, so no - there's no good way to do that.

But - more importantly - you're not using the second value if the first value is `false` so it doesn't really matter. The only reason we include it is for type stability (which is why we should be using `zero(T)`). I think a comment explaining this fact that we ignore the second value if the first one is false is sufficient here.